File: python_backend/app/domain/entities/recording.py
"""Recording domain entity (B6 — Pro+).

A 1:1 sidecar to ``Session``: when a user with the recording feature
unlocked ends a session with ``is_recording=True``, the audio chunks are
materialised into storage and a ``Recording`` row links the storage key to
the session for later playback / signed URLs.

Pure data — no framework or infrastructure dependencies."""
from dataclasses import dataclass
from datetime import datetime
from typing import Literal, Optional


AudioFormat = Literal["pcm", "mp3", "opus", "webm", "wav", "ogg", "m4a"]


@dataclass
class Recording:
    """An audio recording attached to a session.

    ``session_id`` is the PK (1 recording max per session — uploads upsert).
    ``audio_path`` is the opaque storage key (S3 object key in prod, local
    filesystem path in dev). ``expires_at`` is computed at upload time from
    the user's ``auto_delete_recordings_days`` preference; ``None`` means
    "never auto-delete"."""

    session_id: str
    audio_path: str
    audio_format: AudioFormat
    audio_duration_seconds: int
    audio_size_bytes: int
    expires_at: Optional[datetime]
    created_at: datetime

File: apps/web/src/domain/entities/recording.ts
/**
 * Recording domain entity — backend `RecordingResponse` mirror (B6).
 *
 * Pro+ tier feature. Each Recording is anchored to a Session via
 * `sessionId` (PK on the backend). The audio file lives in object
 * storage; the URL is fetched on-demand via a signed-URL endpoint and
 * expires after `expires_seconds` (~1h default).
 *
 * `expiresAt` (entity-level) is the auto-delete date based on the
 * user's `auto_delete_recordings_days` preference. NULL means the
 * recording is kept indefinitely (the user has explicitly disabled
 * auto-cleanup in /app/settings).
 *
 * Times are kept as ISO 8601 UTC strings — presentation parses with
 * the user locale only when rendering.
 */

export type AudioFormat =
  | "pcm"
  | "mp3"
  | "opus"
  | "webm"
  | "wav"
  | "ogg"
  | "m4a";

export interface Recording {
  sessionId: string;
  audioFormat: AudioFormat;
  audioDurationSeconds: number;
  audioSizeBytes: number;
  /** Auto-delete date (null = never expires). */
  expiresAt: string | null;
  createdAt: string;
}

/**
 * Recording joined with the parent Session metadata for the listing
 * endpoint (`GET /api/recordings`). Each row carries enough info for the
 * recordings list to render without a second round-trip per item.
 */
export interface RecordingWithSession extends Recording {
  sessionTitle: string | null;
  sessionScenario: string;
  sessionStartedAt: string;
  sessionEndedAt: string | null;
}
